pkg/azure/credential/ucpcredentials.go

```diff
@@ -102,48 +103,91 @@ func (c *UCPCredential) refreshCredentials(ctx context.Context) error {
 		return err
 	}
 
-	if s.ClientID == "" || s.ClientSecret == "" || s.TenantID == "" {
-		return errors.New("invalid azure service principal credential info")
-	}
+	switch s.Kind {
+	case datamodel.AzureServicePrincipalCredentialKind:
+		azureServicePrincipalCredential := s.ServicePrincipal
+		if azureServicePrincipalCredential.ClientID == "" || azureServicePrincipalCredential.ClientSecret == "" || azureServicePrincipalCredential.TenantID == "" {
+			return errors.New("invalid azure service principal credential info")
+		}
+
+		// Do not instantiate new client unless the secret is rotated.
+		if c.credential != nil &&
+			c.credential.ServicePrincipal != nil &&
+			c.credential.ServicePrincipal.ClientSecret == azureServicePrincipalCredential.ClientSecret &&
+			c.credential.ServicePrincipal.ClientID == azureServicePrincipalCredential.ClientID &&
+			c.credential.ServicePrincipal.TenantID == azureServicePrincipalCredential.TenantID {
+			c.refreshExpiry()
+			return nil
+		}
+
+		logger.Info("Retreived Azure Credential - ClientID: " + azureServicePrincipalCredential.ClientID)
+
+		// Rotate credentials by creating new ClientSecretCredential.
+		var opt *azidentity.ClientSecretCredentialOptions
+		if c.options.ClientOptions != nil {
+			opt = &azidentity.ClientSecretCredentialOptions{
+				ClientOptions: *c.options.ClientOptions,
+			}
+		}
+
+		azCred, err := azidentity.NewClientSecretCredential(azureServicePrincipalCredential.TenantID, azureServicePrincipalCredential.ClientID, azureServicePrincipalCredential.ClientSecret, opt)
+		if err != nil {
+			return err
+		}
+
+		c.tokenCred = azCred
+		c.credential = s
 
-	// Do not instantiate new client unless the secret is rotated.
-	if c.credential != nil && c.credential.ClientSecret == s.ClientSecret &&
-		c.credential.ClientID == s.ClientID && c.credential.TenantID == s.TenantID {
 		c.refreshExpiry()
 		return nil
-	}
+	case datamodel.AzureWorkloadIdentityCredentialKind:
+		azureWorkloadIdentityCredential := s.WorkloadIdentity
+		if azureWorkloadIdentityCredential.ClientID == "" || azureWorkloadIdentityCredential.TenantID == "" {
+			return errors.New("invalid azure workload identity credential info")
```

Comment:
nit: I'd suggest providing an explicit error message here

```suggestion
			return errors.New("empty clientID or tenantID provided for Azure workload identity")
```